Solution1/Solution1.Tests/Integration/WeatherApiTests.cs

```diff
@@ -0,0 +1,178 @@
+        public async Task GetForecastByCityName_Success()
+        {
+            // Arrange            
+            var startDateTime = DateTime.Now.Date;
+            var httpClient = GetClient();
+            var request = new HttpRequestMessage(HttpMethod.Get, $"/api/weather/{cityName}/{counDays}");
+
+            //Act
+            var response = await httpClient.SendAsync(request);
+
+            //Assert
+            Assert.NotNull(response);
+            Assert.NotNull(response.Content);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            
+            var forecast = JsonSerializer.Deserialize<ForecastWeatherDTO>(await response.Content.ReadAsStringAsync(), _serializerOptions);
+
+            Assert.NotNull(forecast);
+            Assert.NotNull(forecast.CityName);
+            Assert.Equal(forecast.CityName, cityName);
+            Assert.NotNull(forecast.WeatherForPeriod);
+            Assert.Equal(forecast.WeatherForPeriod.Count, counDays+1);
+            
+            forecast.WeatherForPeriod.ForEach(x =>
+            {
+                Assert.NotNull(x);
+                Assert.NotNull(x.Comment);
+                Assert.Contains(x.Comment, comments);
+                Assert.Equal(startDateTime, x.DateTime);
+                startDateTime = startDateTime.AddDays(1);
+            });
+        }
+
+        [Theory]
+        [InlineData("aaaaaaaaaaaaaaaaaaaaa", HttpStatusCode.BadRequest)]
+        [InlineData("", HttpStatusCode.NotFound)]
+        public async Task GetWeatherByCityName_EnterInvalidData_HandlingException(string cityName, HttpStatusCode httpStatusCode)
+        {
+            // Arrange
+            var request = new HttpRequestMessage(HttpMethod.Get, $"/api/weather/{cityName}");
+            var httpClient = GetClient();
+
+            //Act
+            var response = await httpClient.SendAsync(request);
+
+            //Assert
+            Assert.NotNull(response);
+            Assert.Equal(httpStatusCode, response.StatusCode);            
+        }
+
+        [Theory]
+        [InlineData("Minsk", -1, HttpStatusCode.BadRequest)]
+        [InlineData("Minsk", 10, HttpStatusCode.BadRequest)]
+        [InlineData("aaaaaaaaaaaaaaaaaaaaa", 3, HttpStatusCode.BadRequest)]
+        [InlineData("aaaaaaaaaaaaaaaaaaaaa", 7, HttpStatusCode.BadRequest)]
+        [InlineData("", 10, HttpStatusCode.NotFound)]
+        [InlineData("", 3, HttpStatusCode.NotFound)]
+
+        public async Task GetForecastByCityName_EnterInvalidData_HandlingException(string cityName, int countDays, HttpStatusCode httpStatusCode)
+        {
+            // Arrange
+            var request = new HttpRequestMessage(HttpMethod.Get, $"/api/weather/{cityName}/{countDays}");
+            var httpClient = GetClient();
+
+            //Act
+            var response = await httpClient.SendAsync(request);
+
+            //Assert
+            Assert.NotNull(response);
+            Assert.Equal(httpStatusCode, response.StatusCode);
```

Comment:
check also exact error messages count, and exact message text

Reply:
check response data is null

Reply:
+